File: 2323-minimum-bit-flips-to-convert-number/minimum-bit-flips-to-convert-number.js
/**
 * @param {number} start
 * @param {number} goal
 * @return {number}
 */
var minBitFlips = function(start, goal) {
    let xor = start ^ goal;
    let res = 0;
    while(xor > 0) {
        // Gives the LSB of the binary rep of the num
        res += (xor & 1);
        // Remove the LSB to get the next one
        xor >>= 1;
    }
    return res;
};

File: 2323-minimum-bit-flips-to-convert-number/minimum-bit-flips-to-convert-number.java
class Solution {
    public int minBitFlips(int start, int goal) {
        int value = start ^ goal;
        int count = 0;
        while(value != 0) {
            count += (value & 1);
            value = value >> 1;
        }
        return count;
    }
}